numpy/_core/multiarray.py

```diff
@@ -1362,11 +1362,11 @@ def shares_memory(a, b, max_work=None):
         of candidate solutions to consider). The following special
         values are recognized:
 
-        max_work=MAY_SHARE_EXACT  (default)
+        max_work=-1 (formerly MAY_SHARE_EXACT)
             The problem is solved exactly. In this case, the function returns
             True only if there is an element shared between the arrays. Finding
             the exact solution may take extremely long in some cases.
-        max_work=MAY_SHARE_BOUNDS
+        max_work=0 (formerly MAY_SHARE_BOUNDS)
             Only the memory bounds of a and b are checked.
```

Comment:
Could mention here that this is the same as `may_share_memory`.  Because really, nobody needs to ever pass these two explicitly.